doc/guides/env_vars.rst

```diff
@@ -0,0 +1,144 @@
+.. _env_vars:
+
+Environment Variables
+=====================
+
+Various pages in this documentation refer to setting Zephyr-specific
+environment variables. This page describes how.
+
+Setting Variables
+*****************
+
+Option 1: Just Once
+===================
+
+To set the environment variable :envvar:`MY_VARIABLE` to ``foo`` for the
+lifetime of your current terminal window:
+
+.. code-block:: console
+
+   # Linux and macOS
+   export MY_VARIABLE=foo
+
+   # Windows
+   set MY_VARIABLE=foo
+
+.. warning::
+
+  This is best for experimentation. If you close your terminal window, use
+  another terminal window or tab, restart your computer, etc., this setting
+  will be lost forever.
+
+  Using options 2 or 3 is recommended if you want to keep using the setting.
+
+Option 2: In all Terminals
+==========================
+
+On macOS and Linux, add the ``export MY_VARIABLE=foo`` line to your shell's
+startup script in your home directory. For Bash, this is usually
+:file:`~/.bashrc`.  The setting will then be available whenever you open a new
```

Comment:
On macOS it's typically `~/.bash_profile`, I would mention this here.

Reply:
Great, thanks!

Reply:
One of too many stackoverflow duplicates trying and failing to summarize this giantic mess: https://superuser.com/questions/183870/difference-between-bashrc-and-bash-profile

Reply:
I just noticed this comment -- what would you like me to do?

Reply:
I think changing just this sentence would be great:
```diff
- The setting will then be available whenever you open a new terminal window.
+ Changes in these startup scripts don't affect shell instances already started.
```

Reply:
Done (with a suggestion to open a new window added after a semicolon)

Reply:
I think this paragraph still gives the user too much confidence that everything will just work the same on any Linux distribution. There wouldn't be so many duplicates of https://superuser.com/questions/183870/difference-between-bashrc-and-bash-profile if it were the case, just check the number of links there or perform a quick search.
Merely adding some adverb ("generally" or "in most cases",...) to the second sentence would be enough to temper it.

Reply:
will change to: " try opening a new terminal window to get the new settings."

Reply:
I dove deeper into the /`etc/profile/.bashrc/bash_profile/`etc. abyss to understand why it was so difficult to unset Clear Linux' default CFLAGS (PR #17029) It's even worse than I expected. Not because of bash itself but because of what the Linux distributions make of it. Here's for instance how Fedora and Clear Linux source default files. Ubuntu is of course completely different again. Longer version at https://superuser.com/a/1452523/111302

```
ssh fedora29
 └─ -bash # login shell
      ├── /etc/profile
      |    ├─ /etc/profile.d/*.sh
      |    ├─ /etc/profile.d/sh.local
      |    └─ /etc/bashrc
      ├── ~/.bash_profile
      |    └─ ~/.bashrc
      |          └─ /etc/bashrc
      |
      |
      └─ $ bash  # non-login shell
            └─ ~/.bashrc
                 └─ /etc/bashrc
                       └─ /etc/profile.d/*.sh

ssh clearlinux
 └─ -bash # login shell
      ├── /usr/share/defaults/etc/profile
      |    ├─ /usr/share/defaults/etc/profile.d/*
      |    ├─ /etc/profile.d/*
      |    └─ /etc/profile
      ├── ~/.bash_profile
      |
      |
      └─  $ bash   # non-login shell
           ├─ /usr/share/defaults/etc/bash.bashrc
           |      ├─ /usr/share/defaults/etc/profile
           |      |    ├─ /usr/share/defaults/etc/profile.d/*
           |      |    ├─ /etc/profile.d/*
           |      |    └─ /etc/profile
           |      └─ /etc/profile
           └─ ~/.bashrc
```